Synthetixio/v3ui
# Release ERC7412 changes (#79)

* add base to network selector

* wip

* Upgrade cannon

* Preset `competition` for base-goerli

* Add base-goerli contract imports

* Lock down latest contracts version release to avoid pulling dev-releases

* Temporarily ts-ignore the type incompatibility

* Solc is a peer deps from cannon router so need to be added

* Ran  yarn upgrade-contracts

* yarn dedupe

* Remove unused hook

* Add pythnetwork/pyth-evm-js

* Create useCollateralPrices

* Add useCollateralPrices

* Dont fetch prices in `loadCollateralTypes`

* Use useCollateralPrices

* Parse erc7412 calls`

* Make withERC7412.ts a proper package

* withERC7412 can take an array of tx's

* Add some ERC721 errors

* Correct multicallAddress

* Fix incorrect balances for non SNX collateral

* Make the return type a big more stricts.
- Make sure gasLimit is set
- Created some type helpers too

* useDeposit flow erc7412 ready

* Help ts

* Add calculateCRatio helper

* Its better to call useCollateralPrices for caching.
- Also add some logs

* Dont use CoreProxy.multicall

* Refactor loadPosition to not use CoreProxy.multicall

* Collateral value no longer part of call position

* Make sure erc7412Call helper have support for multiple txs

* Add erc7412-tester page

* ERC7412 for borrow

* Deps and minor ts issue

* ERC7412 for Repay
- and deps

* No math with undefined

* ERC7412 for undelegate

* erc7412Call for useAccountCollateral

* Fix "Add collateral" ui.
- Need to differentiate SNX from other collateral

* Pass down liquidity position to VaultRow

* Refactor loadPosition to returns txData and decoder instead of doing tx

* Make loadPrices returns calls and decoder

* Add keyBy helper

* Load prices with multicall

* Add label to account collateral

* Deps

* Update multicall address for base-goerli to be the trusted forwarder

* Use multicall from contracts, this should make sure we dont break non base networks

* Fix collateral section tests

* Remove testnet flag

* Safely grabbed the error data

* isMarketCapacityLocked requires erc7412Call

* Fetch price when fetching single liquidity position

* Only fetch the active account collateral on deposit screen

* Deps

* Cache read calls for 5s

* Lint

* Only refetch syntehtix is collateral type is snx

* Fix symbol issue

* Only use multicall is we have trusted forwarder
- Also we make sure to use the right WETH based on the connected network

* use useAccountSpecificCollateral if only fetching for one collateral address

* Remove tester page

* Deps

* Clearer syntax when refetching

* Better error msg

* Refactor maxWithdraw

* Fix unit tests

* typo

* frame rpc fix

* satsua url

* other frame issue + btc icon

* revert

* Always use JsonRpcProvider

* Dont fetch availableCollateral twice when deposit

* Lint

* This is not needed anymore, withERC7412 handles this

* Less refetching of liq pos when depositing

* Pass liq position to borrow

* LiqPosition to contain accountCollateral
- Also refetch of liq position only to happen on close

* Preload collateral prices (#77)

* Add OracleManagerProxy

* Add use useAllCollateralPriceIds

* Add pyth fetcher lib

* Prefetch borrow

* Prefetch for deposit

* Remove duplicates.
- There might be collateral using the same oracle id, we still only need one price update

* Pass availableCollateral to Repay modal

* Allowance need to be refetched after repay, since if no infinity approval it will decrease

* Prefetch for useRepay

* Include collateral address

* Prefetch for undelegate

* Prefetch for useLiquidityPosition
- Make sure erc7412Call filters away price calls, even when prefetching
- Also some logs cleanup

* Preload for usePoolConfigurations

* deps

* useVaultsData erc7412 ready with preload

* Deps

* Lint

* Preload useLiquidityPositions

* Deps

* Dont fetch price updates on networks missing trustedForwarder

* Fix borrow and deposit

* Fix repay
- Repay need sUSD available collateral

* Missed a refetch for repay

* Fix undelegate

* Deps

* Lint

* fix deposit

* wait repay

* We dont want manual waits as we want to assert that data is reloaded before the repay is "done"
- Removing for now, if it fails I will investigate. It's green locally

* We should not return early if account id is missing as this hook is used to create accounts too

---------



* Deps and Remove Space

* same react query version everywhere

* Also upgrade zod

* refetchQueries -> invalidateQueries

* Remove duplicated base-goerli and make it enabled

* ensure AccountSpecificCollateral gets refetched properly after withdrawal properly

* dedupe

* consistent trailing slash

* Remove incorrect todo

* Remove unused function

* Prettier

* Update staking/lib/useAllCollateralPriceIds/useAllCollateralPriceIds.ts



* Update staking/lib/usePoolConfiguration/usePoolConfiguration.ts



---------
